File: Remove All Adjacent Duplicates in String II.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var removeDuplicates = function(s, k) {
    let st = [];
    

for(let i=0;i<s.length;i++){
     let c = s[i];

     if(st.length > 0 && st[st.length-1][0] === c){
           st[st.length-1][1]++
        
  }
   else{
     st.push([c,1]);
      
  }

  if(st.length>0 && st[st.length-1][1] === k){
      st.pop();
  }


}
let ans =""
     for(let i=0;i<st.length;i++){
         let ch=st[i][0];
         let cnt = st[i][1];

        for(let j=0;j<cnt;j++){
            ans+=ch;
        }
     }

     return ans;

    
};

File: Java/Remove All Adjacent Duplicates in String II.java
import java.util.Stack;

class Solution {
    public String removeDuplicates(String s, int k) {
        Stack<int[]> st = new Stack<>();
        
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            
            if (!st.isEmpty() && st.peek()[0] == c) {
                st.peek()[1]++;
            } else {
                st.push(new int[]{c, 1});
            }
            
            if (!st.isEmpty() && st.peek()[1] == k) {
                st.pop();
            }
        }
        
        StringBuilder ans = new StringBuilder();
        
        for (int[] pair : st) {
            char ch = (char) pair[0];
            int cnt = pair[1];
            
            for (int j = 0; j < cnt; j++) {
                ans.append(ch);
            }
        }
        
        return ans.toString();
    }
}